### PR TITLE
Simple Buffer

### PR DESCRIPTION
**Pull request by BobMak**
Suggesting to remove the self.total_timesteps; though unlikely, we could want to run learn multiple times, in which case the total timesteps would really be greater than the one specified;

suggesting to remove the HPARAM_ATTRS from the base class and interpret kwargs as hparams (consistent with what we have rn for logging);

Merged our simple replay buffer; simplified logging code; refactored eval step logging.

### CODE REVIEW

**Comment by JacobHA** [comment]
Regarding self.total_timesteps: how do you want to calculate epsilon decay then? How do you think it should be done if training is resumed (should epsilon continue where it left off, or restart its decay schedule).

Should log_history have a step instead of *val?

DQN will probably need some bug fixes because of our buffer changes

**Comment by BobMak** [comment]
I think it makes sense to have any scheduled hparams change as a function of learn() timesteps, since there is usually a max/min range in mind. A different range can be setup if one wants to call learn multiple times. This could support having multiple schedules for various stages of training.

I think the default expected behavior for the step is to increase regardless of the number of learn().

I'll add a variable for local and global learn steps. For the first learn, they will be equal, but later, the local steps will reset.

File: BaseAgent.py [hunk 1]
```diff
@@ -117,13 +123,11 @@ def __init__(self,
     def log_hparams(self, hparam_dict):
         # Log the agent's hyperparameters:
         for logger in self.loggers:
-            # hparam_values = {attr: self.__dict__[attr] for attr in self.HPARAM_ATTRS}
             logger.log_hparams(hparam_dict)
 
-    def log_h(self, param, val):
+    def log_history(self, param, *val):
```
Comment (JacobHA) [inline BaseAgent.py]: this is the *val -> step question I had, should it instead be 
```def log_history(self, param, val, step):``` ?

**Comment by JacobHA** [comment]
that sounds reasonable. what is a decent way to have such local and global steps? I guess I'll see what you add :)

**Review by JacobHA — COMMENTED** [review]
LGTM
:oil_drum:

**Pull request merged**